lib/course_planner/mailer/user_email.ex

```diff
@@ -1,35 +1,23 @@
 defmodule CoursePlanner.Mailer.UserEmail do
   @moduledoc """
-  This module for creating emails
+  Module responsible for building and sending email
   """
-  import Swoosh.Email
+  use Phoenix.Swoosh, view: CoursePlanner.EmailView, layout: {CoursePlanner.LayoutView, :email}
 
-  @type recipient_name :: String.t
-  @type recipient_email :: String.t
-  @type recipient_info :: {recipient_name, recipient_email}
-  @type target_group :: :to | :cc | :bcc
+  @notifications %{
+    user_modified: %{subject: "Your profile was updated", template: "user_updated.html"}
+  }
 
-  def welcome(user) do
-    {"Dr B Banner", "hulk.smash@example.com"}
-    |> create_empty_email()
-    |> add_recepients(:to, {user.name, user.email})
-    |> subject("Hello, Avengers!")
-    |> html_body("<h1>Hello #{user.name}</h1>")
-    |> text_body("Hello #{user.name}\n")
-  end
-
-  @spec add_recepients(Swoosh.Email, target_group, [recipient_info]) :: Swoosh.Email
-  def add_recepients(email, target_group, recipient_list) when is_list(recipient_list) do
-    Map.update!(email, target_group, &(&1 = recipient_list))
-  end
-  @spec add_recepients(Swoosh.Email, target_group, recipient_info) :: Swoosh.Email
-  def add_recepients(email, target_group, recipient) do
-    Map.update!(email, target_group, &(&1 = [recipient]))
-  end
-
-  @spec create_empty_email(recipient_info) :: Swoosh.Email
-  def create_empty_email(sender_info) do
-    new()
-    |> from(sender_info)
+  def build_email(%{name: _, email: email}, _) when is_nil(email), do: {:error, :invalid_recipient}
+  def build_email(%{name: name, email: email}, notification_type) do
+    case @notifications[notification_type] do
```

Comment:
I believe that you should use pattern matching whenever you can in Elixir if it doesn't get too clumped up, and it has better performance (not that it matters too much here).
I also believe I can `%{name: _, email: nil}`